doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,270 @@
+### Extract the Result for Multiple Function-Solution Pairs
+
+If you are solving a system of equations with multiple unknown functions,
+{func}`~.dsolve` will return a nested list of equalities, the outer list
+representing each solution and the inner list representing each function. While
+you can extract results by specifying the index of each function ("slicing" each
+solution), we recommend an approach which is robust with respect to function
+ordering. The following converts each solution into a dictionary so you can
+easily extract the result for the desired function. It uses standard Python
+techniques such as a loops or comprehensions, in a nested fashion.
+
+```py
+>>> from sympy import symbols, Eq, Function, dsolve
+>>> y, z = symbols("y z", cls=Function)
+>>> x = symbols("x")
+>>> eqs = [Eq(y(x).diff(x)**2, z(x)**2), Eq(z(x).diff(x), z(x))]
+>>> solutions = dsolve(eqs); solutions
+[[Eq(y(x), C1 - C2*exp(x)), Eq(z(x), C2*exp(x))], [Eq(y(x), C1 + C2*exp(x)), Eq(z(x), C2*exp(x))]]
+>>> solutions_list = [] # nested list approach
+>>> for solution in solutions:
+...     solution_dict = {}
+...     for fn in solution:
+...             solution_dict.update({fn.lhs: fn.rhs})
+...     solutions_list.append(solution_dict)
+>>> solutions_list
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list = [{fn.lhs:fn.rhs for fn in solution} for solution in solutions]
+>>> solutions_list # nested comprehension approach
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list[0][y(x)]
+C1 - C2*exp(x)
+```
+
+### Work With Arbitrary Constants
+
+When you want to simplify arbitrary constants such as `C1`, `C2`, `C3`, etc.,
+use {func}`~.constantsimp`. For example, if you add two ODE solutions together,
+there may be an opportunity to simplify the constants:
+
+```py
+>>> from sympy import Function, dsolve, Derivative, symbols
+>>> from sympy.abc import x
+>>> from sympy.solvers.ode.ode import constantsimp
+>>> y = Function('y')
+>>> result = dsolve(Derivative(y(x), x, x) + 9*y(x), y(x)); result
+Eq(y(x), C1*sin(3*x) + C2*cos(3*x))
+>>> sol = result.rhs; sol
+C1*sin(3*x) + C2*cos(3*x)
+>>> ex = sol + (3 * sol / 2); ex
+5*C1*sin(3*x)/2 + 5*C2*cos(3*x)/2
+>>> C1, C2, C3 = symbols("C1 C2 C3")
+>>> constantsimp(ex, {C1, C2, C3})
+C1*sin(3*x) + C2*cos(3*x)
+```
+
+After simplifying the arbitrary constants, those constants may be
+non-consecutive ordered. For example the simplified result may contain `C1` and
+`C3` but not `C2`. You can fix this using {func}`~.constant_renumber`:
+
+```py
+>>> from sympy import symbols, cos, sin
+>>> from sympy.solvers.ode.ode import constantsimp, constant_renumber
+>>> C1, C2, C3 = symbols("C1 C2 C3")
+>>> expr = C1*cos(x) + C2*cos(x) + C3*sin(x); expr
+C1*cos(x) + C2*cos(x) + C3*sin(x)
+>>> simplified = constantsimp(expr, {C1, C2, C3}); simplified
+C1*cos(x) + C3*sin(x)
+>>> constant_renumber(simplified)
+C1*sin(x) + C2*cos(x)
+```
```

Comment:
I was trying to come up with realistic examples of when you would work with arbitrary constants. Perhaps it's my ignorance, but I didn't find the examples in [`constantsimp`](https://docs.sympy.org/dev/modules/solvers/ode.html?highlight=constantsimp#sympy.solvers.ode.constantsimp) or [`constant_renumber`](https://docs.sympy.org/dev/modules/solvers/ode.html?highlight=constantsimp#sympy.solvers.ode.ode.constant_renumber) very compelling because they don't come from `dsolve` results (they just start by creating expressions involving `C1`, `C2`, `C3`) despite `constantsimp` being "written specifically to work with [dsolve()](https://docs.sympy.org/dev/modules/solvers/ode.html?highlight=constantsimp#sympy.solvers.ode.dsolve)". I understand the statement "There is no need to specify the dependent function, such as f(x), because it already has the independent symbol, x, in it." but think the examples would follow more naturally if they involved functions as `dsolve` typically returns such as $\cos(x)$.

For `constantsimp`, the best I came up with was adding one solution to a multiple of itself. Is there a more realistic situation where arbitrary constants could be simplified?

For `constant_renumber`, is there some example where applying `constantsimp` leads to removal of an arbitrary constant?

Reply:
I don't think we should discuss either of these functions. They can be useful for end users in some situations, like if you are solving an ODE that comes from another ODE, so that you end up with constants from both solutions, but those situations are not typical. 

`constantsimp` and `constant_renumber` are already applied automatically to the result of `dsolve` (unless you specify `simplify=False`; and unlike `solve`, there's rarely a reason to need to do that).